### PR TITLE
Added Merge Sort in Perl

### LINKED ISSUE #1502: Add Merge Sort in Perl
See requirements: https://sample-programs.therenegadecoder.com/projects/merge-sort/
Add snippet here: https://github.com/TheRenegadeCoder/sample-programs/tree/master/archive/p/perl

### PR DESCRIPTION
**Pull request by maheshwarkuchana**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][1]! For simplicity, please make sure that your pull request includes one and only one contribution.

## Complete the Applicable Sections Below

Find which section best describes your pull request and make sure you fill it out. To start, let us know which issue you've fixed.

- [x] I fixed #1502 

### Code Snippets

- [x] I named the pull request using `Created Merge Sort in Perl` format

### CODE REVIEW

File: archive/p/perl/mergesort.pl [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    for (@x) {
+        $_ = !@a            ? shift @b
+           : !@b            ? shift @a
+           : $a[0] <= $b[0] ? shift @a
+           :                  shift @b;
+    }
+    @x;
+}
+ 
+my @a = (4, 65, 2, -31, 0, 99, 83, 782, 1);
```
Comment (jrg94) [inline archive/p/perl/mergesort.pl]: Any chance you could pull these from the command line? See project description: https://sample-programs.therenegadecoder.com/projects/merge-sort/

File: archive/p/perl/mergesort.pl [hunk 2]
```diff
@@ -0,0 +1,18 @@
+sub merge_sort {
```
Comment (jrg94) [inline archive/p/perl/mergesort.pl]: Can you also rename this to `merge-sort`? That should get it tested.

**Comment by maheshwarkuchana** [comment]
I have changed the file name. But if we change merge_sort to merge-sort it is giving error in perl file.

**Comment by jrg94** [comment]
That's fine! The main issue here is that the program doesn't pull input from the user. Right now, the array is hardcoded.

**Comment by jrg94** [comment]
Going to close this out for now. Feel free to rework and reopen.

**Comment by jrg94** [comment]
Going to briefly reopen this then we'll fix it when it's merged.

**Pull request closed without merging**